packages/plugin-typescript/src/lib/runner/runner.ts

```diff
@@ -0,0 +1,60 @@
+import type {
+  AuditOutput,
+  AuditOutputs,
+  AuditReport,
+  Issue,
+  RunnerFunction,
+} from '@code-pushup/models';
+import type { AuditSlug } from '../types.js';
+import {
+  type DiagnosticsOptions,
+  getTypeScriptDiagnostics,
+} from './ts-runner.js';
+import type { CodeRangeName } from './types.js';
+import { getIssueFromDiagnostic, tSCodeToAuditSlug } from './utils.js';
+
+export type RunnerOptions = DiagnosticsOptions & {
+  expectedAudits: { slug: AuditSlug }[];
+};
+
+export function createRunnerFunction(options: RunnerOptions): RunnerFunction {
+  const { tsconfig, expectedAudits } = options;
+  return async (): Promise<AuditOutputs> => {
+    const diagnostics = await getTypeScriptDiagnostics({ tsconfig });
+    const result: Record<
+      CodeRangeName,
+      Pick<AuditReport, 'slug' | 'details'>
+    > = diagnostics.reduce(
+      (acc, diag) => {
+        const slug = tSCodeToAuditSlug(diag.code);
+        const existingIssues: Issue[] =
+          (acc[slug] && acc[slug].details?.issues) || ([] as Issue[]);
+        return {
+          ...acc,
+          [slug]: {
+            slug,
+            details: {
+              issues: [...existingIssues, getIssueFromDiagnostic(diag)],
+            },
+          },
+        };
+      },
+      {} as unknown as Record<
+        CodeRangeName,
+        Pick<AuditOutput, 'slug' | 'details'>
+      >,
+    );
+
+    return expectedAudits.map(({ slug }) => {
+      const { details } = result[slug as CodeRangeName] ?? {};
```

Comment:
Another unnecessary casting :sweat_smile:

The `AuditSlug` is actually an alias of `CodeRangeName`, so no conversion is needed: 

```suggestion
      const { details } = result[slug] ?? {};
```